okta/api_response.py

```diff
@@ -0,0 +1,135 @@
+import json
+import xmltodict
+
+
+class OktaAPIResponse():
+    """
+    Class for defining the wrapper of an Okta API response.
+    Allows for paginated results to be retrieved easily.
+    """
+
+    def __init__(self, request_executor, req, res_details, response_body={}):
+        self._url = res_details.url
+        self._headers = req["headers"]
+        self._self = None  # Link to first page of results
+        self._body = None  # First page of results
+
+        # Status on if there's a next page of results (based on generator)
+        self._next = None
+
+        # Request Executor for future calls
+        self._request_executor = request_executor
+
+        # Build response body based on content type
+        if "application/xml" == res_details.content_type:
```

Comment:
I would do a `contains` check here as the `xml` response has some extra data in the content-type header